triton/stream.py

```diff
@@ -441,14 +442,19 @@ class AWSStream(Stream):
 
     def __init__(self, name, partition_key, conn=None, region='us-east-1'):
         self.conn = conn or connect_to_region(region)
-        self.name = name
-        self.partition_key = partition_key
+        self.name = ascii_to_unicode_str(name)
+        self.partition_key = ascii_to_unicode_str(partition_key)
         self._shard_ids = None
 
         super(AWSStream, self).__init__()
 
     def _partition_key(self, data):
-        return unicode(data[self.partition_key])
+        try:
+            return ascii_to_unicode_str(data[self.partition_key])
+        except KeyError:
+            #supports a user putting escaped unicode data in, still returns
```

Comment:
What prevents this branch from polluting the original KeyError exception with a unicdoe_to_ascii_str version?

Reply:
fixing now